.ebextensions/20_packages.config

```diff
@@ -8,51 +8,65 @@ packages:
-  03_makeclean:
+  0190_mostly_ugprade_pip:
+    command: "source /opt/python/run/venv/bin/activate && pip install --upgrade pip >> /var/log/deploy.log"
+  0191_check_pip_version_anew:
+    command: "source /opt/python/run/venv/bin/activate && pip --version"
+  0200_install_poetry:
+    command: "source /opt/python/run/venv/bin/activate && pip install poetry >> /var/log/deploy.log"
+  0201_install_poetry:
+    command: "source /opt/python/run/venv/bin/activate && poetry --version"
+  0210_superstitiously_make_clean:
     command: "make clean"
-  04_install_buildout:
-    command: "source /opt/python/run/venv/bin/activate && pip install -U zc.buildout setuptools==36.6.0 && /opt/python/run/venv/bin/buildout bootstrap >> /var/log/deploy.log"
-  043_bootstrap:
+  0217_check_pip_state:
+    command: "source /opt/python/run/venv/bin/activate && pip freeze"
+  0218_uninstall_setuptools:
+    command: "source /opt/python/run/venv/bin/activate && pip uninstall -y setuptools"
+  0219_install_setuptools:
+    command: "source /opt/python/run/venv/bin/activate && pip install setuptools"
+  0220_populate_venv:
+    command: "source /opt/python/run/venv/bin/activate && poetry install"
+  0221_get_aws_ips:
+    command: "make aws-ip-ranges"
+  0400_superstitiously_activate_venv:
     command: "source /opt/python/run/venv/bin/activate"
-  045_ensure_leader_runs_indexing:
-    command: "sed -i 's/indexer =.*/indexer = true/' beanstalk.cfg"
-    leader_only: true
-  046_ensure_leader_runs_mpindexing:
-    command: "sed -i 's/mpindexer.*/mpindexer = true/' beanstalk.cfg"
-    leader_only: true
-  047_install_wheel:
+  0420_still_more_debugging_info:
+    command: "source /opt/python/run/venv/bin/activate && echo $PATH"
+  0470_install_wheel:
     command: "pip install wheel"
-  048_npm_tmp_perms:
+  0480_npm_tmp_perms:
     command: "chown -R ec2-user /tmp"
-  049_app_bundle_perms:
+  0490_app_bundle_perms:
     command: "chown -R ec2-user /opt/python/bundle/"
-  05_secret_key:
-    command: cat /dev/urandom | head -c 256 | base64 > session-secret.b64
-  06_buildout:
-    command: "source /opt/python/run/venv/bin/activate && bin/buildout -c beanstalk.cfg  >> /var/log/deploy.log"
-    ignoreErrors: true
-  07_clear_db_es_contents:
-    command: "bin/clear-db-es-contents production.ini --app-name app --skip-es --env fourfront-mastertest >> /var/log/deploy.log"
+  0500_secret_key:
+    command: "cat /dev/urandom | head -c 256 | base64 > session-secret.b64"
+  0600_generate_production_ini:
+    command: "source /opt/python/run/venv/bin/activate && python deploy/generate_production_ini.py"
+  0696_pip_install_encoded:
+    command: "source /opt/python/run/venv/bin/activate && python setup_eb.py develop"
+  0700_clear_db_es_contents:
+    command: "source /opt/python/run/venv/bin/activate && clear-db-es-contents production.ini --app-name app --skip-es --env fourfront-mastertest >> /var/log/deploy.log"
```

Comment:
We don't have access to `$ENV_NAME` as far as I can tell. You can look at the script and it is definitely something to be improved but what it effectively does is within the script if $ENV_NAME != what was passed in as an argument then it skips the `clear-db-es-contents` step. Since the argument passed in is mastertest it will only run on mastertest.

Reply:
I did a test with
```
0699_env_check:
    command: "echo ENV_NAME=$ENV_NAME"
```
and the log says:
```
Result:
  ENV_NAME=fourfront-cgapdev
```
(I am guessing that if you don't manage to obtain it, you might have done used single quotes, which in bash doesn't expand `$` references. Or something like that.)

Even so, if we changed the argument name to `--if-env` it might be OK for now. Ultimately, when we have more time, I'd rather have the command not be called in the case it's not needed than assume that it shouldn't act. That just seems semantically weird. If we need safeguards, I think we could do better another way.